IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -263,7 +263,9 @@ - (BOOL)saveAccount:(nonnull MSIDAccountCacheItem *)account
     if (previousAccount)
     {
         // Make sure we copy over all the additional fields
-        [account updateFieldsFromAccount:previousAccount];
+        NSMutableDictionary *mergedDictionary = [previousAccount.jsonDictionary mutableCopy];
+        [mergedDictionary addEntriesFromDictionary:account.jsonDictionary];
+        account = [[MSIDAccountCacheItem alloc] initWithJSONDictionary:mergedDictionary error:error];
```

Comment:
nit: check that account creation succeeded before proceeding with saving

Reply:
updated